qemu/tests/ksm_ksmtuned.py

```diff
@@ -31,41 +32,76 @@ def run(test, params, env):
     :param params: Dictionary with test parameters.
     :param env: Dictionary with the test environment.
     """
-    def check_ksm(mem, stress=False):
+    def check_ksm(mem, threshold_reached=False):
         """
         :param mem: Boot guest with given memory, in KB
-        :param stress: Load stress or not
+        :ksmtuned_enabled: ksmtuned threshold is reached or not
         """
+        def heavyload_install():
+            if session.cmd_status(test_install_cmd) != 0:
+                logging.warning("Could not find installed heavyload in guest, "
+                                "will install it via winutils.iso ")
+                winutil_drive = utils_misc.get_winutils_vol(session)
+                if not winutil_drive:
+                    test.cancel("WIN_UTILS CDROM not found.")
+                install_cmd = params["install_cmd"] % winutil_drive
+                session.cmd(install_cmd)
+
+        def check_qemu_used_mem(qemu_pid, mem):
+            qemu_used_page = process.getoutput(get_qemu_used_mem % qemu_pid,
+                                               shell=True)
+            qemu_used_mem = float(qemu_used_page) * pagesize
+            if qemu_used_mem < mem * mem_thres:
+                return False
+            return True
+
         params['mem'] = mem // 1024
         params['start_vm'] = 'yes'
         vm_name = params['main_vm']
         env_process.preprocess_vm(test, params, env, vm_name)
         vm = env.get_vm(vm_name)
-        vm.wait_for_login()
-        if stress and params["os_type"] == "linux":
+        session = vm.wait_for_login()
+        qemu_pid = vm.get_pid()
+        if params["os_type"] == "linux":
             params['stress_args'] = ('--cpu 4 --io 4 --vm 2 --vm-bytes %sM' %
                                      (int(params['mem']) // 2))
             stress_test = VMStress(vm, "stress", params)
             stress_test.load_stress_tool()
+        else:
+            install_path = params["install_path"]
+            test_install_cmd = 'dir "%s" | findstr /I heavyload' % install_path
+            heavyload_install()
+            heavyload_bin = r'"%s\heavyload.exe" ' % install_path
+            heavyload_options = ["/MEMORY 100", "/START"]
```

Comment:
Without duration, do you aim to make stress keep running until `check_qemu_used_mem` timeout ?

Same for linux guest, no unload. I think it is better to add tstep to stop the service after checking qemu mem.

Reply:
Yes, it's intended to run until L90.  We can't be sure about the time it takes. So I didn't set duration. 

Why do we need unload? vm.destroy will be called, every process inside will be terminated.

Reply:
Firstly, it is a normal clean up action in os, use qemu process to control it is not a normal operation,  i think it is same with `session.close`. Secondly, it is test case's response, not framework, this case does not destroy vm but leave it to the framework, it is not a good behaviour too. 
In short, graceful way to clean up is the first choice then rude. 


And,

Reply:
> Firstly, it is a normal clean up action in os, use qemu process to control it is not a normal operation, i think it is same with `session.close`. Secondly, it is test case's response, not framework, this case does not destroy vm but leave it to the 

Actually it's not that same, because `session` is another process outside of qemu process, while the stress only running inside guest and will not leave something else on guest image like file creation something, so that's not that necessary to force a stop before destroy vm. And to destroy vm or not actually is independent with stopping stress or not, from my view that's totally another issue, which has no difference with other test cases that does not start stress.

So, I accept current implementation.

framework, it is not a good behaviour too.
> In short, graceful way to clean up is the first choice then rude.
> 
> And,

Reply:
> Actually it's not that same, because `session` is another process outside of qemu process, while the stress only running inside guest and will not leave something else on guest image like file creation something, so that's not that necessary to force a stop before destroy vm. 

You are right, not same with session. I was convinced that it is not must, but still prefer do clean up more.

> And to destroy vm or not actually is independent with stopping stress or not, from my view that's totally another issue, which has no difference with other test cases that does not start stress.

Yeah, it is another issue that not related with the initial issue about stop stress or not. I comment them together to explain it is better to do clean up in test case instead of framework. Include stop stress, destroy vm and so on ~

@huangyum  As the comment above, i was convinced that it is not must, then up you, thanks!

> 
> So, I accept current implementation.
> 
> framework, it is not a good behaviour too.
> 
> > In short, graceful way to clean up is the first choice then rude.

Reply:
Thanks @vivianQizhu @yanan-fu for the comments.  I don't see the necessity to unload stress before destroy vm. So keep as it is. Thanks.

 (1/2) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.ksm_ksmtuned: PASS (230.40 s)
 (2/2) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.ksm_ksmtuned: PASS (1775.48 s)